src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -365,13 +372,15 @@ private static unsafe void GetCursorPosition(out int left, out int top)
             {
                 // Because the CPR request/response protocol involves blocking until we get a certain
                 // response from the terminal, we want to avoid doing so if we don't know the terminal
-                // will definitely response.  As such, we start with minChars == 0, which causes the
+                // will definitely respond.  As such, we start with minChars == 0, which causes the
                 // terminal's read timer to start immediately.  Once we've received a response for
                 // a request such that we know the terminal supports the protocol, we then specify
                 // minChars == 1.  With that, the timer won't start until the first character is
                 // received.  This makes the mechanism more reliable when there are high latencies
-                // involved in reading/writing, such as when accessing a remote system.
-                Interop.Sys.InitializeConsoleBeforeRead(minChars: (byte)(s_everReceivedCursorPositionResponse ? 1 : 0), decisecondsTimeout: 10);
+                // involved in reading/writing, such as when accessing a remote system. We also extend
+                // the timeout on the very first request to 15 seconds, to account for potential latency
+                // before we know if we will receive a response.
```

Comment:
So if CPR isn't supported, the terminal is going to hang for 15 seconds each time the program runs?  That seems like a really bad experience.

Reply:
Also, is the comment even accurate?  I see `100` below, which would be 10 seconds, not 15, right?

Reply:
Per @wtgodbe , he has a better way to deal with it now.